### ISSUE TITLE
12 errors to run test_modelnet.py

### ISSUE THREAD

**Issue by ozjongwon** [NVIDIAGameWorks/kaolin#432]
I'm new to kaolin and probably I'm doing something stupid. Any helps will be highly appreciated! Thanks!

Installed/configure:

```
# Ubuntu 20.04
sudo apt install nvidia-driver-470 nvidia-cuda-toolkit
conda create --name kaolin python=3.7
conda activate kaolin
conda install jupyter pytorch=1.7.1 cython=0.29.20 scipy Pillow torchvision torchaudio cudatoolkit=10.2 matplotlib -c pytorch
python setup.py develop
```

Then `pytest tests/python/kaolin/io/test_modelnet.py` on the latesthead:

```
git log
commit eacec9615f9b863d3d3a81fb57e839fe6869c250 (HEAD -> master, origin/master, origin/HEAD)
Author: Caenorst <cfujitsang@nvidia.com>
Date:   Tue Aug 24 11:27:54 2021 -0400
```

And got 12 errors:

```
============================= test session starts ==============================
platform linux -- Python 3.7.11, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
rootdir: /home/jc/Work/kaolin
collected 12 items

tests/python/kaolin/io/test_modelnet.py EEEEEEEEEEEE                     [100%]

==================================== ERRORS ====================================
____ ERROR at setup of TestModelNet.test_basic_getitem[0-train-categories0] ____

self = <test_modelnet.TestModelNet object at 0x7f3d935837d0>
categories = ['bathtub'], split = 'train'

    @pytest.fixture(autouse=True)
    def modelnet_dataset(self, categories, split):
        return ModelNet(root=MODELNET_PATH,
                        categories=categories,
>                       split=split)

tests/python/kaolin/io/test_modelnet.py:47: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <kaolin.io.modelnet.ModelNet object at 0x7f3cf14ab490>
root = '/data/ModelNet', categories = ['bathtub'], split = 'train'

    def __init__(self, root, categories=None, split='train'):
        assert split in ['train', 'test'], f'Split must be either train or test ,but got {split}.'
    
        self.root = Path(root)
        self.paths = []
        self.labels = []
    
        if not os.path.exists(root):
>           raise ValueError(f'ModelNet was not found at "{root}.')
E           ValueError: ModelNet was not found at "/data/ModelNet.

kaolin/io/modelnet.py:44: ValueError
____ ERROR at setup of TestModelNet.test_basic_getitem[0-train-categories1] ____

self = <test_modelnet.TestModelNet object at 0x7f3cf14bed10>
categories = ['desk'], split = 'train'

    @pytest.fixture(autouse=True)
    def modelnet_dataset(self, categories, split):
        return ModelNet(root=MODELNET_PATH,
                        categories=categories,
>                       split=split)

tests/python/kaolin/io/test_modelnet.py:47: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <kaolin.io.modelnet.ModelNet object at 0x7f3cf14bee50>
root = '/data/ModelNet', categories = ['desk'], split = 'train'

    def __init__(self, root, categories=None, split='train'):
        assert split in ['train', 'test'], f'Split must be either train or test ,but got {split}.'
    
        self.root = Path(root)
        self.paths = []
        self.labels = []
    
        if not os.path.exists(root):
>           raise ValueError(f'ModelNet was not found at "{root}.')
E           ValueError: ModelNet was not found at "/data/ModelNet.

kaolin/io/modelnet.py:44: ValueError
____ ERROR at setup of TestModelNet.test_basic_getitem[0-train-categories2] ____

self = <test_modelnet.TestModelNet object at 0x7f3cf1478510>
categories = ['bathtub', 'desk'], split = 'train'

    @pytest.fixture(autouse=True)
    def modelnet_dataset(self, categories, split):
        return ModelNet(root=MODELNET_PATH,
                        categories=categories,
>                       split=split)

tests/python/kaolin/io/test_modelnet.py:47: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <kaolin.io.modelnet.ModelNet object at 0x7f3cf1478a10>
root = '/data/ModelNet', categories = ['bathtub', 'desk'], split = 'train'

    def __init__(self, root, categories=None, split='train'):
        assert split in ['train', 'test'], f'Split must be either train or test ,but got {split}.'
    
        self.root = Path(root)
        self.paths = []
        self.labels = []
    
        if not os.path.exists(root):
>           raise ValueError(f'ModelNet was not found at "{root}.')
E           ValueError: ModelNet was not found at "/data/ModelNet.

kaolin/io/modelnet.py:44: ValueError
____ ERROR at setup of TestModelNet.test_basic_getitem[0-test-categories0] _____

self = <test_modelnet.TestModelNet object at 0x7f3cf1443a90>
categories = ['bathtub'], split = 'test'

    @pytest.fixture(autouse=True)
    def modelnet_dataset(self, categories, split):
        return ModelNet(root=MODELNET_PATH,
                        categories=categories,
>                       split=split)

tests/python/kaolin/io/test_modelnet.py:47: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <kaolin.io.modelnet.ModelNet object at 0x7f3cf1443ad0>
root = '/data/ModelNet', categories = ['bathtub'], split = 'test'

    def __init__(self, root, categories=None, split='train'):
        assert split in ['train', 'test'], f'Split must be either train or test ,but got {split}.'
    
        self.root = Path(root)
        self.paths = []
        self.labels = []
    
        if not os.path.exists(root):
>           raise ValueError(f'ModelNet was not found at "{root}.')
E           ValueError: ModelNet was not found at "/data/ModelNet.

kaolin/io/modelnet.py:44: ValueError
____ ERROR at setup of TestModelNet.test_basic_getitem[0-test-categories1] _____

self = <test_modelnet.TestModelNet object at 0x7f3cf142a790>
categories = ['desk'], split = 'test'

    @pytest.fixture(autouse=True)
    def modelnet_dataset(self, categories, split):
        return ModelNet(root=MODELNET_PATH,
                        categories=categories,
>                       split=split)

tests/python/kaolin/io/test_modelnet.py:47: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <kaolin.io.modelnet.ModelNet object at 0x7f3cf142a890>
root = '/data/ModelNet', categories = ['desk'], split = 'test'

    def __init__(self, root, categories=None, split='train'):
        assert split in ['train', 'test'], f'Split must be either train or test ,but got {split}.'
    
        self.root = Path(root)
        self.paths = []
        self.labels = []
    
        if not os.path.exists(root):
>           raise ValueError(f'ModelNet was not found at "{root}.')
E           ValueError: ModelNet was not found at "/data/ModelNet.

kaolin/io/modelnet.py:44: ValueError
____ ERROR at setup of TestModelNet.test_basic_getitem[0-test-categories2] _____

self = <test_modelnet.TestModelNet object at 0x7f3cf1478c10>
categories = ['bathtub', 'desk'], split = 'test'

    @pytest.fixture(autouse=True)
    def modelnet_dataset(self, categories, split):
        return ModelNet(root=MODELNET_PATH,
                        categories=categories,
>                       split=split)

tests/python/kaolin/io/test_modelnet.py:47: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <kaolin.io.modelnet.ModelNet object at 0x7f3cf1478d10>
root = '/data/ModelNet', categories = ['bathtub', 'desk'], split = 'test'

    def __init__(self, root, categories=None, split='train'):
        assert split in ['train', 'test'], f'Split must be either train or test ,but got {split}.'
    
        self.root = Path(root)
        self.paths = []
        self.labels = []
    
        if not os.path.exists(root):
>           raise ValueError(f'ModelNet was not found at "{root}.')
E           ValueError: ModelNet was not found at "/data/ModelNet.

kaolin/io/modelnet.py:44: ValueError
___ ERROR at setup of TestModelNet.test_basic_getitem[-1-train-categories0] ____

self = <test_modelnet.TestModelNet object at 0x7f3cf142a310>
categories = ['bathtub'], split = 'train'

    @pytest.fixture(autouse=True)
    def modelnet_dataset(self, categories, split):
        return ModelNet(root=MODELNET_PATH,
                        categories=categories,
>                       split=split)

tests/python/kaolin/io/test_modelnet.py:47: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <kaolin.io.modelnet.ModelNet object at 0x7f3cf142a410>
root = '/data/ModelNet', categories = ['bathtub'], split = 'train'

    def __init__(self, root, categories=None, split='train'):
        assert split in ['train', 'test'], f'Split must be either train or test ,but got {split}.'
    
        self.root = Path(root)
        self.paths = []
        self.labels = []
    
        if not os.path.exists(root):
>           raise ValueError(f'ModelNet was not found at "{root}.')
E           ValueError: ModelNet was not found at "/data/ModelNet.

kaolin/io/modelnet.py:44: ValueError
___ ERROR at setup of TestModelNet.test_basic_getitem[-1-train-categories1] ____

self = <test_modelnet.TestModelNet object at 0x7f3cf143e350>
categories = ['desk'], split = 'train'

    @pytest.fixture(autouse=True)
    def modelnet_dataset(self, categories, split):
        return ModelNet(root=MODELNET_PATH,
                        categories=categories,
>                       split=split)

tests/python/kaolin/io/test_modelnet.py:47: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <kaolin.io.modelnet.ModelNet object at 0x7f3cf143e650>
root = '/data/ModelNet', categories = ['desk'], split = 'train'

    def __init__(self, root, categories=None, split='train'):
        assert split in ['train', 'test'], f'Split must be either train or test ,but got {split}.'
    
        self.root = Path(root)
        self.paths = []
        self.labels = []
    
        if not os.path.exists(root):
>           raise ValueError(f'ModelNet was not found at "{root}.')
E           ValueError: ModelNet was not found at "/data/ModelNet.

kaolin/io/modelnet.py:44: ValueError
___ ERROR at setup of TestModelNet.test_basic_getitem[-1-train-categories2] ____

self = <test_modelnet.TestModelNet object at 0x7f3cf142ad90>
categories = ['bathtub', 'desk'], split = 'train'

    @pytest.fixture(autouse=True)
    def modelnet_dataset(self, categories, split):
        return ModelNet(root=MODELNET_PATH,
                        categories=categories,
>                       split=split)

tests/python/kaolin/io/test_modelnet.py:47: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <kaolin.io.modelnet.ModelNet object at 0x7f3cf1441090>
root = '/data/ModelNet', categories = ['bathtub', 'desk'], split = 'train'

    def __init__(self, root, categories=None, split='train'):
        assert split in ['train', 'test'], f'Split must be either train or test ,but got {split}.'
    
        self.root = Path(root)
        self.paths = []
        self.labels = []
    
        if not os.path.exists(root):
>           raise ValueError(f'ModelNet was not found at "{root}.')
E           ValueError: ModelNet was not found at "/data/ModelNet.

kaolin/io/modelnet.py:44: ValueError
____ ERROR at setup of TestModelNet.test_basic_getitem[-1-test-categories0] ____

self = <test_modelnet.TestModelNet object at 0x7f3cf143e9d0>
categories = ['bathtub'], split = 'test'

    @pytest.fixture(autouse=True)
    def modelnet_dataset(self, categories, split):
        return ModelNet(root=MODELNET_PATH,
                        categories=categories,
>                       split=split)

tests/python/kaolin/io/test_modelnet.py:47: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <kaolin.io.modelnet.ModelNet object at 0x7f3cf143ee90>
root = '/data/ModelNet', categories = ['bathtub'], split = 'test'

    def __init__(self, root, categories=None, split='train'):
        assert split in ['train', 'test'], f'Split must be either train or test ,but got {split}.'
    
        self.root = Path(root)
        self.paths = []
        self.labels = []
    
        if not os.path.exists(root):
>           raise ValueError(f'ModelNet was not found at "{root}.')
E           ValueError: ModelNet was not found at "/data/ModelNet.

kaolin/io/modelnet.py:44: ValueError
____ ERROR at setup of TestModelNet.test_basic_getitem[-1-test-categories1] ____

self = <test_modelnet.TestModelNet object at 0x7f3cf145e250>
categories = ['desk'], split = 'test'

    @pytest.fixture(autouse=True)
    def modelnet_dataset(self, categories, split):
        return ModelNet(root=MODELNET_PATH,
                        categories=categories,
>                       split=split)

tests/python/kaolin/io/test_modelnet.py:47: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <kaolin.io.modelnet.ModelNet object at 0x7f3cf145e310>
root = '/data/ModelNet', categories = ['desk'], split = 'test'

    def __init__(self, root, categories=None, split='train'):
        assert split in ['train', 'test'], f'Split must be either train or test ,but got {split}.'
    
        self.root = Path(root)
        self.paths = []
        self.labels = []
    
        if not os.path.exists(root):
>           raise ValueError(f'ModelNet was not found at "{root}.')
E           ValueError: ModelNet was not found at "/data/ModelNet.

kaolin/io/modelnet.py:44: ValueError
____ ERROR at setup of TestModelNet.test_basic_getitem[-1-test-categories2] ____

self = <test_modelnet.TestModelNet object at 0x7f3cf1451d10>
categories = ['bathtub', 'desk'], split = 'test'

    @pytest.fixture(autouse=True)
    def modelnet_dataset(self, categories, split):
        return ModelNet(root=MODELNET_PATH,
                        categories=categories,
>                       split=split)

tests/python/kaolin/io/test_modelnet.py:47: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <kaolin.io.modelnet.ModelNet object at 0x7f3cf1451ad0>
root = '/data/ModelNet', categories = ['bathtub', 'desk'], split = 'test'

    def __init__(self, root, categories=None, split='train'):
        assert split in ['train', 'test'], f'Split must be either train or test ,but got {split}.'
    
        self.root = Path(root)
        self.paths = []
        self.labels = []
    
        if not os.path.exists(root):
>           raise ValueError(f'ModelNet was not found at "{root}.')
E           ValueError: ModelNet was not found at "/data/ModelNet.

kaolin/io/modelnet.py:44: ValueError
=========================== short test summary info ============================
ERROR tests/python/kaolin/io/test_modelnet.py::TestModelNet::test_basic_getitem[0-train-categories0]
ERROR tests/python/kaolin/io/test_modelnet.py::TestModelNet::test_basic_getitem[0-train-categories1]
ERROR tests/python/kaolin/io/test_modelnet.py::TestModelNet::test_basic_getitem[0-train-categories2]
ERROR tests/python/kaolin/io/test_modelnet.py::TestModelNet::test_basic_getitem[0-test-categories0]
ERROR tests/python/kaolin/io/test_modelnet.py::TestModelNet::test_basic_getitem[0-test-categories1]
ERROR tests/python/kaolin/io/test_modelnet.py::TestModelNet::test_basic_getitem[0-test-categories2]
ERROR tests/python/kaolin/io/test_modelnet.py::TestModelNet::test_basic_getitem[-1-train-categories0]
ERROR tests/python/kaolin/io/test_modelnet.py::TestModelNet::test_basic_getitem[-1-train-categories1]
ERROR tests/python/kaolin/io/test_modelnet.py::TestModelNet::test_basic_getitem[-1-train-categories2]
ERROR tests/python/kaolin/io/test_modelnet.py::TestModelNet::test_basic_getitem[-1-test-categories0]
ERROR tests/python/kaolin/io/test_modelnet.py::TestModelNet::test_basic_getitem[-1-test-categories1]
ERROR tests/python/kaolin/io/test_modelnet.py::TestModelNet::test_basic_getitem[-1-test-categories2]
============================== 12 errors in 2.92s ==============================
```

**Comment by Caenorst** [comment]
You can ignore the tests for dataset, we usually exclude them using the env variable:
```
export ci='true'
```

if you really want to test the dataset then ModelNet have to be in `/data/ModelNet/` path.

**Comment by ozjongwon** [comment]
Thanks for the explanation!

(I can't remember details, but) When I followed the tutorial, I had an error so I wanted to check if my installation and configuration are sound by running tests.

I confirm that today, just after `git pull` the tutorial works perfectly - I guess it was broken when I tried.

Thank you again!

**Comment by ozjongwon** [comment]
Oops, I lied! I still have the same error. I'll try to run the test after putting data into `/data/ModelNet/ `.

```
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-7-a8a733d72573> in <module>
     76         category='optimized_mesh',
     77         vertices_list=[vertices_batch[0]],
---> 78         materials_list=pbr_material
     79     )

~/Work/kaolin/kaolin/visualize/timelapse.py in add_mesh_batch(self, iteration, category, vertices_list, faces_list, uvs_list, face_uvs_idx_list, face_normals_list, materials_list)
    194 
    195             io.usd.add_mesh(stage, f'/{mesh_name}', vertices, faces, uvs,
--> 196                             face_uvs_idx, face_normals, time=iteration)
    197             if materials is not None:
    198                 if isinstance(materials, io.materials.Material):

~/Work/kaolin/kaolin/io/usd.py in add_mesh(stage, scene_path, vertices, faces, uvs, face_uvs_idx, face_normals, time)
    503         usd_mesh.GetFaceVertexIndicesAttr().Set(faces_list, time=time)
    504     if vertices is not None:
--> 505         vertices_list = vertices.cpu().float().numpy()
    506         usd_mesh.GetPointsAttr().Set(Vt.Vec3fArray.FromNumpy(vertices_list), time=time)
    507     if uvs is not None:

RuntimeError: Can't call numpy() on Tensor that requires grad. Use tensor.detach().numpy() instead.
```

**Comment by ozjongwon** [comment]
Yes, all tests run successfully. So I guess the above `RuntimeError: Can't call numpy() on Tensor that requires grad. Use tensor.detach().numpy() instead.` is another problem.

Please feel free to close this ticket(then I'll open a new ticket), or any hints to fix the issue I have(then keep this open).

Thanks.

**Comment by Caenorst** [comment]
Hi @ozjongwon actually there was a regression bug due to a recent optimization on USD exporter. I've quickly fixed the issue on master and v0.9.1 please give it a try!

Thank you for reporting!
